container/container_unix.go

```diff
@@ -219,7 +219,7 @@ func (container *Container) UnmountIpcMounts(unmount func(pth string) error) {
 			logrus.Error(err)
 			warnings = append(warnings, err.Error())
 		} else if shmPath != "" {
-			if err := unmount(shmPath); err != nil && !os.IsNotExist(err) {
+			if err := unmount(shmPath); err != nil && !os.IsNotExist(err) && err != unix.EINVAL {
```

Comment:
On Linux, EINVAL may mean that umount2() was called with an invalid flag value in flags.
We need to use Mounted.